main.go

```diff
@@ -43,42 +48,105 @@ const (
-	dbTimeout                     time.Duration
-	debug                         bool
-	useSSH                        bool
-	sshHost                       string
-	sshPort                       string
-	sshUser                       string
-	sshKeyFilePath                string
-	chanSize                      int
-}
+type (
+	options struct {
+		// Required vars
+		appEnv                                         string
+		pgHost, pgPort, pgUser, pgPassword, pgDatabase string
+		dbInstanceConnectionName                       string
+		apiKeys                                        map[string]bool
+		// Optional vars
+		port                          string
+		maxRelayBatchSize             int
+		maxRelayBatchDuration         time.Duration
+		maxServiceRecordBatchSize     int
+		maxServiceRecordBatchDuration time.Duration
+		dbTimeout                     time.Duration
+		debug                         bool
+		chanSize                      int
+	}
+
+	// DB config structs
+	DBConfig interface {
+		GetDriver(ctx context.Context) (driver *postgresdriver.PostgresDriver, cleanup func() error, err error)
+	}
+	CloudSQLConfig struct {
+		options
+	}
+	TestDBConfig struct {
+		options
+	}
+)
 
 func gatherOptions() options {
-	return options{
-		connectionString:              environment.MustGetString(connectionString),
-		apiKeys:                       environment.MustGetStringMap(apiKeys, ","),
+	appEnv := environment.GetString(appEnv, "production")
+
+	options := options{
+		appEnv: appEnv,
+		// Required vars
+		pgUser:     environment.MustGetString(pgUser),
+		pgPassword: environment.MustGetString(pgPassword),
+		pgDatabase: environment.MustGetString(pgDatabase),
+		apiKeys:    environment.MustGetStringMap(apiKeys, ","),
+		// Optional vars
 		port:                          environment.GetString(port, defaultPort),
 		maxRelayBatchSize:             int(environment.GetInt64(maxRelayBatchSize, defaultBatchSize)),
 		maxRelayBatchDuration:         time.Duration(environment.GetInt64(maxRelayBatchDuration, defaultBatchDuration)) * time.Second,
 		maxServiceRecordBatchSize:     int(environment.GetInt64(maxServiceRecordBatchSize, defaultBatchSize)),
 		maxServiceRecordBatchDuration: time.Duration(environment.GetInt64(maxServiceRecordBatchDuration, defaultBatchDuration)) * time.Second,
 		dbTimeout:                     time.Duration(environment.GetInt64(dbTimeout, defaultDBTimeout)) * time.Second,
 		debug:                         environment.GetBool(debug, defaultDebug),
-		useSSH:                        environment.GetBool(useSSH, defaultUseSSH),
-		sshHost:                       environment.GetString(sshHost, ""),
-		sshPort:                       environment.GetString(sshPort, ""),
-		sshUser:                       environment.GetString(sshUser, ""),
-		sshKeyFilePath:                environment.GetString(sshKeyFilePath, ""),
 		chanSize:                      int(environment.GetInt64(chanSize, defaultChanSize)),
 	}
+
+	switch appEnv {
+	case "production":
```

Comment:
The environment being "production" or else should not be visible to the application. Please use string processing (or any other method you see fit) to prevent this detail from leaking into the code.